src/helpers/referal.js

```diff
@@ -57,36 +56,18 @@ __private.list = function (filter, cb) {
         return setImmediate(cb, 'Invalid limit. Maximum is 100');
     }
 
-    const orderBy = OrderBy(
-        (filter.orderBy || 'reward_time:desc'), {
-            sortFields: sql.sortFields
-        }
-    );
-
-    if (orderBy.error) {
-        return setImmediate(cb, orderBy.error);
-    }
-
-    library.db.query(sql.list({
-        where,
-        sortField: orderBy.sortField,
-        sortMethod: orderBy.sortMethod
-    }), params).then((rows) => {
-        let count = 0;
-        if (rows && rows.length !== 0) {
-            count = Number(rows[0].total_rows);
-        }
-
-        const data = {
-            rewards: rows,
-            count
-        };
-
-        return setImmediate(cb, null, data);
-    }).catch((err) => {
-        library.logger.error(err.stack);
-        return setImmediate(cb, 'Rewards#list error');
-    });
+    /** Resolved Count Issue on Response with Updated Code */
+
+        library.db.query(sql.getReferralRewardHistory, {
+            introducer_address: params.introducer_address,
+            offset: params.offset,
+            limit: params.limit
+        })
+        .then(row => setImmediate(cb, null, {
+            rewards: row,
+            count: row[0].rewards_count
```

Comment:
row can be empty

Reply:
Here I used query method for which it will return the row with empty array and count will be 0 which we are using on Frontend so that we are showing pagination on the basis of count.